src/index.js

```diff
@@ -22,70 +22,57 @@ function optimist(fn) {
     let {optimist, innerState} = separateState(state);
     var newOptimist = [], started = false, committed = false;
     optimist.forEach(function (entry) {
-      if (started) {
-        if (
-          entry.beforeState &&
-          matchesTransaction(entry.action, action.optimist.id)
-        ) {
+      if (matchesTransaction(entry.action, action.optimist.id)) {
+        if (entry.beforeState) {
           committed = true;
-          newOptimist.push({action: entry.action});
-        } else {
-          newOptimist.push(entry);
+          entry = {action: entry.action} // Strip beforeState - we're never going to need to revert to it.
+        }
+      }
+      else {
+        if (entry.beforeState) {
+          started = true; // We're in an open transaction, start recording all entries
         }
-      } else if (
-        entry.beforeState &&
-        !matchesTransaction(entry.action, action.optimist.id)
-      ) {
-        started = true;
+      }
+      if (started) {
         newOptimist.push(entry);
-      } else if (
-        entry.beforeState &&
-        matchesTransaction(entry.action, action.optimist.id)
-      ) {
-        committed = true;
       }
     });
     if (!committed) {
-      console.error('Cannot commit transaction with id "my-transaction" because it does not exist');
+      console.error('Cannot commit transaction with id "' + action.optimist.id + '" because it does not exist');
     }
     optimist = newOptimist;
     return baseReducer(optimist, innerState, action);
   }
   function revertReducer(state, action) {
-    let {optimist, innerState} = separateState(state);
-    var newOptimist = [], started = false, gotInitialState = false, currentState = innerState;
+    let {optimist, ...ignore} = separateState(state);
+    var newOptimist = [], started = false, currentState = undefined;
```

Comment:
That's true for redux, but these are just functions, and i don't want to make that assumption in general.  For example, what if redux-optimist was only used on some small, deeply nested reducer, and a higher level reducer chose not to pass down the `@@INIT` action.  We should do our best to make this generic, rather than tying it to the specifics of redux.